src/prompt.ts

```diff
@@ -0,0 +1,84 @@
+import * as inquirer from "inquirer";
+import * as _ from "lodash";
+
+import * as FirebaseError from "./error";
+
+export interface Choice {
+  checked: boolean;
+  label: string;
+  name: string;
+}
+export type Question = inquirer.Question;
+
+/**
+ * prompt is used to prompt the user for values. Specifically, any `name` of a
+ * provided question will be checked against the `options` object. If `name`
+ * exists as a key in `options`, it will *not* be prompted for. If `options`
+ * contatins `nonInteractive = true`, then any `question.name` that does not
+ * have a value in `options` will cause an error to be returned. Once the values
+ * are queried, the values for them are put onto the `options` object, and the
+ * answers are returned.
+ * @param options The options object passed through by Command.
+ * @param questions `Question`s to ask the user.
+ * @return The answers, keyed by the `name` of the `Question`.
+ */
+export async function prompt(options: { [key: string]: any }, questions: Question[]): Promise<any> {
+  const prompts = [];
+  for (const question of questions) {
+    if (question.name && !options[question.name]) {
+      prompts.push(question);
+    }
+  }
+
+  if (prompts.length && options.nonInteractive) {
+    const missingOptions = _.uniq(_.map(prompts, "name")).join(", ");
+    throw new FirebaseError(
+      `Missing required options (${missingOptions}) while running in non-interactive mode`,
+      {
+        children: prompts,
+        exit: 1,
+      }
+    );
+  }
+
+  const answers = await inquirer.prompt(prompts);
+  // lodash's forEach's call back is (value, key); this is not a typo.
+  _.forEach(answers, (v, k) => {
+    options[k] = v;
+  });
+  return answers;
+}
+
+/**
+ * Quick version of `prompt` to ask a single question.
+ * @param question The question (of life, the universe, and everything).
+ * @return The value as returned by `inquirer` for that quesiton.
+ */
+export async function promptOnce(question: Question): Promise<any> {
+  question.name = question.name || "question";
+  const answers = await prompt({}, [question]);
+  return answers[question.name];
+}
+
+/*
+ * Converts a list of `Choice` objects into a list of `Question` objects which
+ * are compatible with `prompt` (and `promptOnce`).
+ */
+export function convertLabeledListChoices(choices: Choice[]): Question[] {
+  return choices.map((choice: any) => {
+    return { checked: choice.checked, name: choice.label };
```

Comment:
Is there a way in JS to create the Question object here instead of using object literal like this? Something like `return new Question({...})` would be much more readable.

Reply:
Also, I'm confused why we're using `choice.label` for name instead of `choice.name`. I went through the code in `src/commands/init.js` and am still not understanding the need for both this and the `listLabelToValue` functions.

Reply:
It's removed.